hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieFileGroup.java

```diff
@@ -62,14 +59,16 @@
   /**
    * The last completed instant, that acts as a high watermark for all getters
    */
-  private final Optional<HoodieInstant> lastInstant;
+  private final com.google.common.base.Optional<HoodieInstant> lastInstant;
 
   public HoodieFileGroup(String partitionPath, String id, HoodieTimeline timeline) {
     this.partitionPath = partitionPath;
     this.id = id;
     this.fileSlices = new TreeMap<>(HoodieFileGroup.getReverseCommitTimeComparator());
     this.timeline = timeline;
-    this.lastInstant = timeline.lastInstant();
+    // Guava Optional is serializable.
+    // TODO : varadarb -> Open another PR to replace java.util.Optional with Guava'a everywhere internally
+    this.lastInstant = com.google.common.base.Optional.fromNullable(timeline.lastInstant().orElse(null));
```

Comment:
Just like here, if we roll out our own optional, composing code requires converting from one type to another which may be necessary for an end user if they want to wrap into a guava optional.

Reply:
I like this idea +1